### PR TITLE
feat(CommonContext): add new context

### PR DESCRIPTION
**Pull request by cpotey**

### CODE REVIEW

File: impl.js [hunk 1]
```diff
@@ -163,7 +163,7 @@ function RPCServiceImplementation(service, impl, transforms = []) {
           }
         });
 
-        applyContextDefaults(data);
+        applyContextDefaults(data, child.resolvedRequestType.fields.context.resolvedType.fullName);
```
Comment (jamessharp) [inline impl.js:166]: Can we pull this out as a constant further up. Also there may not be a `context` field (or in fact any of the interim objects may be undefined) so will have to handle that case (we'll get an exception being thrown here)

File: impl.js [hunk 2]
```diff
@@ -241,7 +241,7 @@ function generateRequestId() {
   return crypto.randomBytes(7).toString('base64');
 }
 
-function applyContextDefaults(data) {
+function applyContextDefaults(data, contextType) {
   // Add in data to the context
   var context = Object.assign({}, data.context || {});
```
Comment (jamessharp) [inline impl.js:250]: Better handle the case where there isn't a context type (i.e. do nothing)

File: test/index.js [hunk 3]
```diff
@@ -296,6 +356,50 @@ const testImpl = {
     };
   }),
 
+  helloCommonContext: sinon.spy(function({ name }) {
+    return {
```
Comment (jamessharp) [inline test/index.js:361]: I think you can just return an empty object here - all the casting etc is being tested elsewhere so you're not adding anything

**Review by jamessharp — CHANGES_REQUESTED** [review]

File: impl.js (hunk 2 above)
Comment (cpotey) [reply impl.js:250]: What's the best solution here?
My initial thought was to add an else to the if/else if logic inside, but I don't know what it should be.

Comment (cpotey) [reply impl.js:250]: Or should it simply be the if/else logic has a `contextType && x` on them too?

Comment (jamessharp) [reply impl.js:250]:
```javascript
if (!contextType) {
  return;
}
```

File: impl.js (hunk 1 above)
Comment (cpotey) [reply impl.js:166]: Struggled with this one, lots of null if I try and do anything with `child.resolvedRequestType.fields.context.resolvedType.fullName` or variations of - not sure what's happening there. It doesn't seem to get anything until you start going into the if/else functions. Is it only accessible inside the `wrappedImpl` functions?

**Pull request merged**